### ISSUE TITLE
Proposal for a better color scheme in report output

### ISSUE THREAD

**Issue by gasche** [OCamlPro/opam-builder#26]
I have mentioned several times that I was not satisfied with the current error scheme of the opam-builder reports, and that question resurfaced during the OCaml Workshop. Here is a proposal of new color scheme:
- forget the difference between light and dark green, just use the same green level for all packages where all available versions compile. On the other hand, use a specific "bad" color (for example Red) for packages that are not installable at all (they build under none of the monitored OCaml versions).
- remove the Red background of the "Deps" cases, that is use the exact same background (white/transparent) as for packages that are marked as unavailable for that version
- turn the orange "Fail" scheme into a red background

Rationale are the following:
- shade of greens: currently it depends on whether the package is available on all monitored OCaml versions. It's not very useful, debatable, and does not clearly scale to supporting more and more OCaml versions.
- Red "Deps" vs. Orange "Fail": The real problem that opam-builder is good at solving is builds that fail. This should be the central issue that gets the attention-catching color. On the other hand, there are good reasons why a package could be uninstallable due to its dependencies, and marking a package unavailable on an OCaml version just because some of its dependencies also are uninstallable -- this is a non-modular situation. On the other hand, we may want to be somehow alerted when a package is uninstallable not because of OCaml version constraints on its dependencies, but because there is a conflict between some of its dependencies (that maybe was not apparent at the time its metadata was written); I think that my suggestion to mark in red the packages that have no installable versions should cover this case.

**Comment by gasche** [comment]
Note: I wouldn't mind coming with a pull request to change this myself, but for this I would rather avoid having to do a full build of the opam-repository first. Could you ( @lefessan ) give instruction on where to find a tarball of a representative build output, and how to go from that and a clone of the opam-builder repository to a working HTML output to test our patches?

(Bonus points if it is possible to skip the build and installation of the opam fork for this workflow.)

**Comment by lefessan** [comment]
@gasche You can find such an archive here:
`http://fabrice.lefessant.net/builder/reports-20160926.tar.gz`

You can do the following:

```
mkdir tmp
cd tmp
tar zxf ~/Download/reports-20160926.tar.gz
mkdir html
 ~/PATH_TO_OPAMBUILDER/_obuild/opam-builder/opam-builder.asm --import */reports
```

You can then view the reports in the `html` directory:

```
xdg-open html/report-last.html
```

Note that you can generate files faster by just providing one `reports` directory instead of all of them.

**Comment by gasche** [comment]
Below is a step-by-step debug/log/guide to html report generation from the information given by Fabrice above (thanks!).

I cloned the repository, installed the opam dependencies (see the `./opam` file), ran `./configure` and `make`, and all that worked just fine -- note that I did not build a forked version of opam or anything like that.

Then I tried the invocation as described, and the first error was

```
$ ~/Prog/opam-builder/_obuild/opam-builder/opam-builder.asm --no-build --import 4.04.0+beta2/reports 
Error: CAML_LD_LIBRARY_PATH is set
```

If I used `unset CAML_LD_LIBRARY_PATH` first, I get:

```
$ unset CAML_LD_LIBRARY_PATH; ~/Prog/opam-builder/_obuild/opam-builder/opam-builder.asm --no-build --import 4.04.0+beta2/reports 
/home/gasche/.opam/4.03.0/bin/ocamlc is in your PATH. You should clean it !
```

setting `PATH=""` for the `opam-builder` invocation then failed with

```
$ unset CAML_LD_LIBRARY_PATH; PATH="" ~/Prog/opam-builder/_obuild/opam-builder/opam-builder.asm --no-build --import 4.04.0+beta2/reports 
CheckImport.import
Loading "4.04.0+beta2/reports/20160926223549-05ae6a0-4.04.0+beta2.export"
Error Unix.Unix_error(Unix.ENOENT, "mkdir", "html/0install") while importing status, 1st time, waiting for 5s and restart.
```

(This does not kill the builder, I have to use Control-C) This is because I forgot to do `mkdir html` first. Then it works.

```
$ mkdir -p html; unset CAML_LD_LIBRARY_PATH; PATH="" ~/Prog/opam-builder/_obuild/opam-builder/opam-builder.asm --no-build --import 4.04.0+beta2/reports 
CheckImport.import
Loading "4.04.0+beta2/reports/20160926223549-05ae6a0-4.04.0+beta2.export"
Generated
Generated
Generated
Generated
All reports generated
```

Note that the tool keeps running after that, I have to use Control-C explicitly. (I tried passing the `--no-build` option to see if it would terminate after report generation, but it does not.)

I think it would be nice to improve a bit on the user-interface for users/contributors that just want to generate reports in this way. I wouldn't mind sending a pull-request to update README to describe this workflow, but I don't understand the reason for the checks I had to work-around above (I guess they are there for robustness of the build phase). Maybe it would be possible to add a `--report-only` option that would disable those checks, and terminate the run after report generation?

**Comment by lefessan** [comment]
The idea behind checking `OCAMLRUNPARAM`, `CAML_LD_LIBRARY_PATH` and `ocamlc` in the `PATH` is to prevent the user from using `opam-builder` on a computer where OCaml is installed in the system, to prevent weird interactions between the builds and the system/user settings.

Indeed, in `tools/opam-builder/main.ml`, the loop after `!arg_import` is an infinite one, it should be easy to add an argument that makes the loop exit after one iteration.

**Comment by gasche** [comment]
@lefessan : it looks like the archive your provided does not contain linting information. Would you upload a version with the linting data as well?

**Comment by gasche** [comment]
The colors were fixed in #27, but it would still be useful to find a better place for the instructions on how to to play with the reporting without doing the building first.
